File: 4-median-of-two-sorted-arrays/median-of-two-sorted-arrays.c
double findMedianSortedArrays(int* nums1, int nums1Size, int* nums2, int nums2Size) {
    int totalSize = nums1Size + nums2Size;
    int* merged = (int*)malloc(totalSize * sizeof(int));
    int i = 0, j = 0, k = 0;

    // Merge the two sorted arrays
    while (i < nums1Size && j < nums2Size) {
        if (nums1[i] <= nums2[j]) {
            merged[k++] = nums1[i++];
        } else {
            merged[k++] = nums2[j++];
        }
    }

    // Add remaining elements from nums1, if any
    while (i < nums1Size) {
        merged[k++] = nums1[i++];
    }

    // Add remaining elements from nums2, if any
    while (j < nums2Size) {
        merged[k++] = nums2[j++];
    }

    double median;
    if (totalSize % 2 == 0) {
        median = (merged[totalSize/2 - 1] + merged[totalSize/2]) / 2.0;
    } else {
        median = merged[totalSize/2];
    }

    free(merged);
    return median;
}

File: 4-median-of-two-sorted-arrays/median-of-two-sorted-arrays.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int m = nums1.length;
        int n = nums2.length;
        int[] merged = new int[m + n];
        
        int i = 0, j = 0, k = 0;
        
        // Merge the two sorted arrays
        while (i < m && j < n) {
            if (nums1[i] <= nums2[j]) {
                merged[k++] = nums1[i++];
            } else {
                merged[k++] = nums2[j++];
            }
        }
        
        // Add remaining elements from nums1, if any
        while (i < m) {
            merged[k++] = nums1[i++];
        }
        
        // Add remaining elements from nums2, if any
        while (j < n) {
            merged[k++] = nums2[j++];
        }
        
        int totalLength = m + n;
        if (totalLength % 2 == 0) {
            // If even number of elements, return average of two middle elements
            return (merged[totalLength/2 - 1] + merged[totalLength/2]) / 2.0;
        } else {
            // If odd number of elements, return the middle element
            return merged[totalLength/2];
        }
    }
}